src/Distribution/Nixpkgs/Haskell/PackageSourceSpec.hs

```diff
@@ -193,18 +194,21 @@ cabalFromDirectory False dir = do
 handleIO :: (Exception.IOException -> IO a) -> IO a -> IO a
 handleIO = Exception.handle
 
+encodeUtf8 :: String -> BS.ByteString
+encodeUtf8 = T.encodeUtf8 . T.pack
+
 hpackDirectory :: FilePath -> MaybeT IO (Bool, Cabal.GenericPackageDescription)
 hpackDirectory dir = do
   mPackage <- liftIO $ Hpack.readPackageConfig Hpack.defaultDecodeOptions {Hpack.decodeOptionsTarget = dir </> Hpack.packageConfig}
   case mPackage of
     Left err -> liftIO $ hPutStrLn stderr ("*** hpack error: " ++ show err ++ ". Exiting.") >> exitFailure
     Right (Hpack.DecodeResult pkg' _ _) -> do
-      let hpackOutput = Hpack.renderPackage [] pkg'
-          hash = printSHA256 $ digestString (digestByName "sha256") hpackOutput
-      case runParseGenericPackageDescription "<hpack output>" (BS.pack hpackOutput) of
+      let hpackOutput = encodeUtf8 $ Hpack.renderPackage [] pkg'
+          hash = printSHA256 $ digest (digestByName "sha256") hpackOutput
+      case runParseGenericPackageDescription "<hpack output>" hpackOutput of
         Left msg -> liftIO $ do
           hPutStrLn stderr "*** hpack output:"
-          hPutStrLn stderr hpackOutput
+          BS.hPutStrLn stderr hpackOutput
```

Comment:
Outputting the plain `ByteString` here is not awesome and will result in gibberish if the locale is not set to UTF8.  However, outputting the `String` as it was before results in an exception if the locale is e.g. `c` and the output contain Unicode code points `> 127`.

Not sure if I like the current code, but also don't know any better option.  We could do a `show` on `hpackOutput` first and then print this to `stderr`, but as the output is potentially very large that would not be awesome neither.

Any suggestions welcome!